src/pilot/dell_nfv.py

```diff
@@ -0,0 +1,376 @@
+                        ConfigOvercloud.ironic_client, node_uuid)
+                stor = client.DRACClient(drac_ip, drac_user, drac_password)
+                # cpu socket information for every compute node
+                sockets = stor.list_cpus()
+                cpu_count = 0
+                for socket in sockets:
+                    if socket.ht_enabled:
+                        cpu_count += socket.cores * 2
+                    else:
+                        raise Exception("Hyperthreading is not enabled in "
+                                        + str(node_uuid))
+                        sys.exit(0)
+                cpu_count_list.append(cpu_count)
+
+            min_cpu_count = min(cpu_count_list)
+            if min_cpu_count not in [40, 48, 56, 64, 72, 128]:
+                raise Exception("CPU count should be one of these"
+                                " values : [40,48,56,64,72,128]"
+                                " But number of cpu is " + str(
+                                    min_cpu_count))
+                sys.exit(0)
+            number_of_host_os_cpu = int(number_of_host_os_cpu)
+            logger.info("host_os_cpus {}".format(
+                cpu_siblings.sibling_info[
+                    min_cpu_count][number_of_host_os_cpu]["host_os_cpu"]))
+            logger.info("vcpus {}".format(
+                cpu_siblings.sibling_info[
+                    min_cpu_count][number_of_host_os_cpu]["vcpu_pin_set"]))
+            return cpu_siblings.sibling_info[
+                min_cpu_count][number_of_host_os_cpu]["vcpu_pin_set"]
+        except Exception as error:
+            message = "Exception {}: {}".format(
+                type(error).__name__, str(error))
+            raise Exception("Failed to calculate "
+                            "Numa Vcpu list {}".format(message))
+
+    @classmethod
+    def calculate_hugepage_count(self, hugepage_size):
+        try:
+
+            pages = []
+
+            for node in ConfigOvercloud.nodes:
+                node_uuid = node.uuid  # uuid of a node
+                node_details = ConfigOvercloud.ironic_client.node.get(
+                    node_uuid)  # details of a node
+                memory_count = node_details.properties['memory_mb']
+                node_properties_capabilities = \
+                    node_details.properties['capabilities'].split(',')[
+                        0].split(':')[1]
+                if 'compute' in node_properties_capabilities:
+                    # Subtracting
+                    # 16384MB = (Host Memory 12GB + Kernel Memory 4GB)
+                    memory_count = (memory_count - 16384)
+                    if hugepage_size == "2MB":
+                        pages.append((memory_count / 2))
+                    if hugepage_size == "1GB":
+                        pages.append((memory_count / 1024))
+            logger.info("hugepage_count {}".format(min(pages)))
+            return min(pages)
```

Comment:
If there is a compute nodes with less than 16 gigs of memory, then this will happily configure a negative number of huge pages.  It seems like we should guard against this scenario.

Reply:
As discussed with Sumedh we have prerequisite RAM size > 128GB
Added condition to check memory should not be less than 128GB